pkg/base/base.go

```diff
@@ -43,16 +43,19 @@ func init() {
 	troubleshootscheme.AddToScheme(scheme.Scheme)
 }
 
-func GetGVKWithNameHash(content []byte) []byte {
+func GetGVKWithNameAndNs(content []byte, baseNS string) string {
 	o := OverlySimpleGVK{}
 
 	if err := yaml.Unmarshal(content, &o); err != nil {
-		return nil
+		return ""
+	}
+
+	namespace := baseNS
```

Comment:
so its possible that both baseNS and o.Metadata.Namespace are empty here but that is ok

Reply:
pretty much my thought
kustomize will match based on the empty ns in that case anyways, so adding some other value would potentially cause errors even if the resources end up colliding during deployment